build.gradle.kts

```diff
@@ -260,6 +258,11 @@ tasks {
     withType<Test>().matching { it.name.startsWith("compatTest") }.configureEach {
         systemProperty("plugin.version", project.version)
     }
+    named<Test>("test").configure {
+        javaLauncher = project.javaToolchains.launcherFor {
```

Comment:
Note: End to end tests are left on Gradle 11, however this could mean we're testing something that the users will never see! (Btw, where is the publish workflow or release steps?)

For E2E We should probably
 * set Gradle to 19
 * leave the compilation as default (i.e. 19)
 * set the runner toolchain to something explicit
    * probably a matrix of 8, 11, 17!

Reply:
https://github.com/gradle-nexus/publish-plugin/pull/254/commits/b2dff42a5bbb6f1d9bf3468392b1c2c3e5381da5